src/fungible/amount.rs

```diff
@@ -141,3 +142,86 @@ impl AddAssign for PreciseAmount {
         }
     }
 }
+
+#[derive(Clone, Debug, Display, PartialEq, Eq)]
+#[display(doc_comments)]
+/// Asset decimal precision exceeds 18 digits
+pub struct DecimalPrecisionOverflowError;
```

Comment:
Rust API guidelines suggest not to add `Error` suffix to names which are obvious for being errors (like containing `Overflow`, `Invalid`) etc)